### PR TITLE
[Serializer] Add new default normalizers

### PR DESCRIPTION
**Pull request by dunglas**
Refactoring of the serializer doc to list new normalizers introduced in 3.1 and enabled by default.

### CODE REVIEW

File: serializer.rst [hunk 1]
```diff
@@ -82,19 +82,30 @@ you need it or it can be used in a controller like the following::
 Adding Normalizers and Encoders
 -------------------------------
 
-Once enabled, the ``serializer`` service will be available in the container
-and will be loaded with two :ref:`encoders <component-serializer-encoders>`
-(:class:`Symfony\\Component\\Serializer\\Encoder\\JsonEncoder` and
-:class:`Symfony\\Component\\Serializer\\Encoder\\XmlEncoder`) and the
-:ref:`ObjectNormalizer normalizer <component-serializer-normalizers>`.
+Once enabled, the ``serializer`` service will be available in the container. It comes with a set of useful :ref:`encoders <component-serializer-encoders>`
+and `normalizers <component-serializer-normalizers>`.
```
Comment (GuilhemN) [inline serializer.rst]: doesn't it miss `:ref:`?

File: serializer.rst [hunk 2]
```diff
@@ -82,19 +82,30 @@ you need it or it can be used in a controller like the following::
...
+* :class:`Symfony\\Component\\Serializer\\Normalizer\\ArrayDenormalizer` to denormalize arrays of objects using a format like `MyObject[]` (note the `[]` suffix)
+
+You can also load custom normalizers and/or encoders by tagging them as
 :ref:`serializer.normalizer <reference-dic-tags-serializer-normalizer>` and
 :ref:`serializer.encoder <reference-dic-tags-serializer-encoder>`. It's also
 possible to set the priority of the tag in order to decide the matching order.
 
 Here is an example on how to load the
-:class:`Symfony\\Component\\Serializer\\Normalizer\\GetSetMethodNormalizer`:
+:class:`Symfony\\Component\\Serializer\\Normalizer\\GetSetMethodNormalizer`, a faster alternative to the `ObjectNormalizer`
```
Comment (GuilhemN) [inline serializer.rst]: should be on two lines

File: serializer.rst [hunk 3]
```diff
@@ -82,19 +82,30 @@ you need it or it can be used in a controller like the following::
 Adding Normalizers and Encoders
 -------------------------------
 
-Once enabled, the ``serializer`` service will be available in the container
-and will be loaded with two :ref:`encoders <component-serializer-encoders>`
-(:class:`Symfony\\Component\\Serializer\\Encoder\\JsonEncoder` and
-:class:`Symfony\\Component\\Serializer\\Encoder\\XmlEncoder`) and the
-:ref:`ObjectNormalizer normalizer <component-serializer-normalizers>`.
+Once enabled, the ``serializer`` service will be available in the container. It comes with a set of useful :ref:`encoders <component-serializer-encoders>`
```
Comment (GuilhemN) [inline serializer.rst]: should be on two lines

Comment (HeahDude) [inline serializer.rst]: Please remember to keep lines shorter than 80 characters.

File: serializer.rst [hunk 4]
```diff
@@ -82,19 +82,30 @@ you need it or it can be used in a controller like the following::
...
+
+As well as the following normalizers:
+
+* :class:`Symfony\\Component\\Serializer\\Normalizer\\ObjectNormalizer` to handle typical data objects
+* :class:`Symfony\\Component\\Serializer\\Normalizer\\DateTimeNormalizer` for objects implementing the :class:`DateTimeInterface` interface
+* :class:`Symfony\\Component\\Serializer\\Normalizer\\DataUriNormalizer` to transform :class:`SplFileInfo` objects in `Data URIs`_
+* :class:`Symfony\\Component\\Serializer\\Normalizer\\JsonSerializableNormalizer` to deal with object implementing the :class:`JsonSerializable` interface
+* :class:`Symfony\\Component\\Serializer\\Normalizer\\ArrayDenormalizer` to denormalize arrays of objects using a format like `MyObject[]` (note the `[]` suffix)
+
+You can also load custom normalizers and/or encoders by tagging them as
```
Comment (HeahDude) [inline serializer.rst]: "Custom normalizers/encoders can also be loaded by tagging..."

File: serializer.rst [hunk 5]
```diff
@@ -82,19 +82,30 @@ you need it or it can be used in a controller like the following::
...
 :ref:`serializer.normalizer <reference-dic-tags-serializer-normalizer>` and
 :ref:`serializer.encoder <reference-dic-tags-serializer-encoder>`. It's also
 possible to set the priority of the tag in order to decide the matching order.
 
 Here is an example on how to load the
-:class:`Symfony\\Component\\Serializer\\Normalizer\\GetSetMethodNormalizer`:
+:class:`Symfony\\Component\\Serializer\\Normalizer\\GetSetMethodNormalizer`, a faster alternative to the `ObjectNormalizer`
+if your data objects always use getters and setters:
```
Comment (HeahDude) [inline serializer.rst]: "When data objects always ..."

**Comment by dunglas** [comment]
Comments fixed.

File: serializer.rst [hunk 6]
```diff
@@ -82,19 +82,37 @@ you need it or it can be used in a controller like the following::
...
+As well as the following normalizers:
+
+* :class:`Symfony\\Component\\Serializer\\Normalizer\\ObjectNormalizer` to
+  handle typical data objects
+* :class:`Symfony\\Component\\Serializer\\Normalizer\\DateTimeNormalizer` for
+  objects implementing the :class:`DateTimeInterface` interface
+* :class:`Symfony\\Component\\Serializer\\Normalizer\\DataUriNormalizer` to
+  transform :class:`SplFileInfo` objects in `Data URIs`_
+* :class:`Symfony\\Component\\Serializer\\Normalizer\\JsonSerializableNormalizer`
+  to deal with object implementing the :class:`JsonSerializable` interface
```
Comment (xabbuh) [inline serializer.rst]: [...] with objects implementing [...]

File: serializer.rst [hunk 7]
```diff
@@ -82,19 +82,37 @@ you need it or it can be used in a controller like the following::
...
+* :class:`Symfony\\Component\\Serializer\\Normalizer\\ArrayDenormalizer` to
+  denormalize arrays of objects using a format like `MyObject[]` (note the `[]` suffix)
+
+Custom normalizers and/or encoders can also be loaded by tagging as
```
Comment (xabbuh) [inline serializer.rst]: [...] tagging them as [...]

**Review by xabbuh — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Thanks Kévin.

**Pull request closed without merging**